src/Bot/Dubtrack.hs

```diff
@@ -63,11 +69,39 @@ songLink (songType -> SongTypeSoundcloud) =
   "Soundcloud links are not supported yet"
 songLink _ = error "This should never happen Kappa"
 
+newtype RoomName = RoomName
+  { unName :: T.Text
+  }
+
+instance IsEntity RoomName where
+  nameOfEntity _ = "RoomName"
+  toProperties reply = Map.fromList [("name", PropertyText $ unName reply)]
+  fromProperties = fmap RoomName . extractProperty "name"
+
+getRoom :: Effect (Entity RoomName)
+getRoom = do
+  reply <- listToMaybe <$> selectEntities P.Proxy (Take 1 All)
+  case reply of
+    Just reply' -> return reply'
+    Nothing -> createEntity P.Proxy $ RoomName "tsoding"
```

Comment:
I don't like the idea of defaulting to any kind of room, because it makes the first experience kinda weird. I think if the room was never set the `!song` command should says something like "The dubtrack room is not set. The mods can set it with !config dubtrack <room>". It is more user friendly if the bot explains how to configure itself. Just like with untrusted link and command messages.

Reply:
This one was more involved but I think the current solution does exactly what you want.